infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespaceFactory.java

```diff
@@ -517,6 +562,74 @@ protected String evalPlaceholders(String namespace, NamespaceResolutionContext c
     return evaluated;
   }
 
+  /**
+   * Stores computed namespace name and it's template into user preferences. Template is required to
+   * track its changes and re-generate namespace in case it didn't matches.
+   */
+  private void recordEvaluatedNamespaceName(String namespace) {
```

Comment:
This is called from `evaluateNamespaceName(NamespaceResolutionContext)` which contains the intended user ID/name, yet we're relying on the environment context to fetch that info here.

I remember that I had problems with this, because the namespace evaluation might be called from contexts that don't have a proper user set up in the environment context, e.g. workspace recovery.

Reply:
you mean i should get it from NamespaceResolutionContext ? ok...